src/Build/Graph/ProjectInterpretation.cs

```diff
@@ -227,6 +233,14 @@ ProjectItemInstance projectReference
             // The properties on the project reference supersede the ones from the MSBuild task instead of appending.
             if (newProperties.Count == 0)
             {
+                // This mimics the _GetProjectReferenceTargetFrameworkProperties task in order to properly reflect what the build graph looks like in
+                // a traversal in which EnableDynamicPlatformResolution is turned on
+                if (ConversionUtilities.ValidBooleanTrue(projectReference.Project.GetPropertyValue(EnableDynamicPlatformResolutionMetadataName)) && String.IsNullOrEmpty(projectReference.GetMetadataValue(SetPlatformMetadataName)))
+                {
+                    var referencedProject = new Project(projectReference.EvaluatedInclude);
```

Comment:
This doesnt seem right.

In any case, you should be using `ProjectInstance` if anything, and you should be using the same settings used elsewhere like the shared evaluation context

Reply:
It doesn't look like you should be doing this evaluation at all. This also seems very incorrect as you're reading `PlatformsMetadataName` as a *Property*, not metadata.